gufe/protocols/protocol_settings.py

```diff
@@ -0,0 +1,381 @@
+from typing import Optional, Dict, Union, List
+
+from pydantic import BaseModel, validator
+from openff.units import unit
+
+
+class SystemSettings(BaseModel):
+    """Settings describing the simulation system settings.
+
+    Attributes
+    ----------
+    nonbonded_method : str
+        Which nonbonded electrostatic method to use, currently only PME
+        is supported.
+    nonbonded_cutoff : float * unit.nanometer
+        Cutoff value for short range interactions.
+        Default 1.0 * unit.nanometer.
+    constraints : str
+        Which bonds and angles should be constrained. Default None.
+    rigid_water : bool
+        Whether to apply rigid constraints to water molecules. Default True.
+    hydrogen_mass : float
+        How much mass to repartition to hydrogen. Default None, no
+        repartitioning will occur.
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    nonbonded_method = 'PME'
+    nonbonded_cutoff = 1.0 * unit.nanometer
+    constraints: Union[str, None] = 'HBonds'  # Usually use HBonds
+    rigid_water = True
+    remove_com = True  # Probably want False here
+    hydrogen_mass: Union[float, None] = None
+
+
+class TopologySettings(BaseModel):
+    """Settings for creating Topologies for each component
+
+    Attributes
+    ----------
+    forcefield : dictionary of list of strings
+      A mapping of each components name to the xml forcefield to apply
+    solvent_model : str
+      The water model to use. Note, the relevant force field file should
+      also be included in ``forcefield``. Default 'tip3p'.
+
+    TODO
+    ----
+    * We can probably just detect the solvent model from the force field
+      defn. In that case we wouldn't have to have ``solvent_model`` here.
+    """
+    # mapping of component name to forcefield path(s)
+    forcefield: Dict[str, Union[List[str], str]]
+    solvent_model = 'tip3p'
+
+
+class AlchemicalSettings(BaseModel):
```

Comment:
It looks like this is a dumping ground for random `perses` alchemical configuration parameters. Perhaps we could meet specifically to discuss how to come up with a better object model to more flexibly represent alchemical protocols, alchemical intermediates, and the like? We've been meaning to do this for a while, and I think it can be done flexibly, extensibly, and elegantly. I fear that just dumping everything into an object like this with attributes that will need to be supported is going to be problematic in the medium- and long terms.